### PR TITLE
Change the way the translation is called, in order to adopt the new translation system

### PR DESCRIPTION
**Pull request by antoninblasco**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Change the way the translation is called, in order to adopt the new translation system.
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| How to test?      | 
| Possible impacts? | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23154)

### CODE REVIEW

File: src/Core/Stock/StockManager.php [hunk 1]
```diff
@@ -319,7 +319,12 @@ protected function sendLowStockAlert($product, $id_product_attribute, $newQuanti
                 Mail::Send(
                     $idLang,
                     'productoutofstock',
-                    Mail::l('Product out of stock', $idLang),
+                    Context::getContext()->getTranslator()->trans(
+                        'Product out of stock',
+                        array(),
```
Comment (Progi1984) [inline src/Core/Stock/StockManager.php]:
```suggestion
                        [],
```

File: src/Core/Stock/StockManager.php [hunk 2]
```diff
@@ -319,7 +319,12 @@ protected function sendLowStockAlert($product, $id_product_attribute, $newQuanti
                 Mail::Send(
                     $idLang,
```
Comment (Progi1984) [inline src/Core/Stock/StockManager.php]:
```suggestion
                    $context->language->local,
```

File: src/Core/Stock/StockManager.php [hunk 3]
```diff
@@ -319,7 +319,12 @@ protected function sendLowStockAlert($product, $id_product_attribute, $newQuanti
...
                     'productoutofstock',
-                    Mail::l('Product out of stock', $idLang),
+                    Context::getContext()->getTranslator()->trans(
```
Comment (Progi1984) [inline src/Core/Stock/StockManager.php]:
```suggestion
                    $context->getTranslator()->trans(
```

File: src/Core/Stock/StockManager.php [hunk 4]
```diff
@@ -319,7 +319,12 @@ protected function sendLowStockAlert($product, $id_product_attribute, $newQuanti
...
+                        'Product out of stock',
```
Comment (Julievrz) [inline src/Core/Stock/StockManager.php]:
```suggestion
                        'Out-of-stock product',
```

**Review by Progi1984 — APPROVED** [review]

**Review by Julievrz — APPROVED** [review]
The wording is good, thank you.

File: src/Core/Stock/StockManager.php [hunk 5]
```diff
@@ -317,9 +317,14 @@ protected function sendLowStockAlert($product, $id_product_attribute, $newQuanti
                 $templateVars['{lastname}'] = $employee->lastname;
 
                 Mail::Send(
-                    $idLang,
+                    $context->language->local,
```
Comment (PierreRambaud) [inline src/Core/Stock/StockManager.php:320]:
```suggestion
                    $context->language->id,
```

local property does not exist :thinking:

**Review by PierreRambaud — COMMENTED** [review]
Hey, thanks for your contribution. Just a little question about `local` I see no other occurrences like that in the Core, is it misspelled?

**Comment by matks** [comment]
Hello, it seems this PR is unfortunately stale. 😞 

I close this PR, following our [review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity". If however you are still interested in it, please ping us :) we can reopen.

Thank you for your contribution.

**Pull request closed without merging**